main.go

```diff
@@ -42,6 +52,69 @@ type Output struct {
 	Expiration      string
 }
 
+type Cache struct {
+	Dir string
+}
+
+func NewCache(dir string) (*Cache, error) {
+	if _, err := os.Open(dir); err != nil {
+		return nil, err
+	}
+
+	fn := path.Join(dir, "validate")
+	if _, err := os.Create(fn); err != nil {
+		return nil, err
+	}
+	os.Remove(fn)
+
+	return &Cache{
+		Dir: dir,
+	}, nil
+}
+
+func (c *Cache) filenameByRole(role string) string {
+	h := sha256.New()
+	h.Write([]byte(role))
+	return fmt.Sprintf("%x", h.Sum(nil))
+}
+
+func (c *Cache) Get(role string) (*Output, bool) {
+	fn := path.Join(c.Dir, c.filenameByRole(role))
+	if f, err := os.Open(fn); err != nil {
+		return nil, false
+	} else {
```

Comment:
unnecessary else

Reply:
fixed.

Reply:
actually I need the `f`

Reply:
The way to do this is:
```go
	f, err := foo()
	if err != nil {
		return nil, false
	}
	[...]
```
Errors are special in Go in a way that you can redeclare the err variable with `:=`
e.g. the following code does not complain about redeclaring err:
```go
	a, err := foo()
	if err != nil {
		return
	}
	fmt.Println(a)
	b, err := foo()
	if err != nil {
		return
	}
	fmt.Println(b)
```
https://go.dev/play/p/k60syEjqGCB